src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -72,6 +72,24 @@ internal sealed class LocalFunctionSymbol : MethodSymbol
                 ReportAttributesDisallowed(param.AttributeLists, _declarationDiagnostics);
             }
 
+            if (syntax.ReturnType.Kind() == SyntaxKind.RefType)
+            {
+                var returnType = (RefTypeSyntax)syntax.ReturnType;
+                if (returnType.ReadOnlyKeyword.Kind() == SyntaxKind.ReadOnlyKeyword)
+                {
+                    _refKind = RefKind.RefReadOnly;
+                    DeclaringCompilation.EnsureIsReadOnlyAttributeExists(_declarationDiagnostics, returnType.Location, modifyCompilation: false);
+                }
+                else
+                {
+                    _refKind = RefKind.Ref;
+                }
+            }
+            else
+            {
+                _refKind = RefKind.None;
+            }
+
             _binder = binder;
```

Comment:
Wrong place for the comment?

---
In reply to: [115860743](https://github.com/dotnet/roslyn/pull/18715#discussion_r115860743) [](ancestors = 115860743)